neuraxle/union.py

```diff
@@ -54,9 +55,19 @@ def __init__(
         self.n_jobs = n_jobs
         self.backend = backend
 
-    def apply(self, method_name: str, *kargs, **kwargs) -> 'BaseStep':
-        BaseStep.apply(self, method_name, *kargs, **kwargs)
-        self.joiner.apply(method_name, *kargs, **kwargs)
+    def apply(self, method_name: str, *kargs, **kwargs) -> Union[Dict, Iterable]:
+        """
+        TODO: fix issue #235
+        :param method_name:
+        :param kargs:
+        :param kwargs:
+        :return:
+        """
+        results = BaseStep.apply(self, method_name, *kargs, **kwargs)
+        other_results = self.joiner.apply(method_name, *kargs, **kwargs)
+        results = join_apply_results(results, other_results)
```

Comment:
I don't think this method is needed, as it should be properly implemented already in what we inherit from: TruncableSteps